accepted/future-releases/nnbd/feature-specification.md

```diff
@@ -616,6 +609,267 @@ It is a warning to use a null aware operator (`?.`, `?[]`, `?..`, `??`, `??=`, o
 `T` if `T` is **strictly non-nullable** .
 
+### Local variables and definite (un)assignment.
+
+As part of the null safety release, errors for local variables are specified to
+take into account **definite assignment** and **definite unassignment** (see
+the section on Definite Assignment below).
+
+In all cases in this section, errors that are described as occurring on reads of
+a variable are intended to apply to all form of reads, including indirectly as
+part of composite assignment operators, as well as via pre and post-fix
+operators.  Similarly, errors that are described as occurring on writes of a
+variable are intended to apply to all form of writes.
+
+It is a compile time error to assign a value to a local variable marked `final`
+unless the variable is **definitely unassigned**, or the variable is marked
+`late`. Conversely, it is not a compile time error to write to a `final` local
+variable if that variable is **definitely unassigned**.
+
+It is a compile time error to assign a value to a local variable marked `final`
+and `late` if the variable is **definitely assigned**.  Conversely, it is not a
+compile time error to write to a `final` local variable if that variable is
+declared `late`, and is not **definitely assigned**.
+
+*Note that a variable is always considered definitely assigned and not
+definitely unassigned if it has an explicit initializer, or an implicit
+initializer as part of a larger construct (e.g. the loop variable in a `for in`
+construct).
+
+It is a compile time error to read a local variable when the variable is
+**definitely unassigned** unless the variable is non-final and has nullable
+type.  This includes variables marked `late` and/or `final`.
+
+The erors specified above are summarized in the following table (using `int` as
+an example non-nullable type).  A variable which has an initializer (explicit or
+implicit) is always considered definitely assigned, and is never considered
+definitely unassigned.
+
+
+Read Errors:
+
+| Declaration form  | Def. Assigned | Neither           | Def. Unassigned |
+| ----------------- | ------------- | ----------------- | --------------- |
+| var x;            | Ok            | Error             | Error           |
+| final x;          | Ok            | Error             | Error           |
+| int x;            | Ok            | Error             | Error           |
+| int? x;           | Ok            | Ok                | Ok              |
+| final int x;      | Ok            | Error             | Error           |
+| final int? x;     | Ok            | Error             | Error           |
+| late var x;       | Ok            | Ok                | Error           |
+| late final x;     | Ok            | Ok                | Error           |
+| late int x;       | Ok            | Ok                | Error           |
+| late final int x; | Ok            | Ok                | Error           |
+
```

Comment:
Should we cover potentially non-nulllable types as well? It is it sufficient to say that they are treated exactly the same as non-nullable types in this regard?